third_party/amd/backend/compiler.py

```diff
@@ -71,6 +72,36 @@ def hash(self):
         return hashlib.sha256(key.encode("utf-8")).hexdigest()
 
 
+class HIPAttrsDescriptor(AttrsDescriptor):
+    __slots__ = ("pointer_range_32")
+
+    @staticmethod
+    def is_within2gb(arg):
+        if hasattr(arg, "ptr_range"):
+            return arg.ptr_range() < 2**31 - 1
+        if "torch.Tensor" in str(type(arg)) and hasattr(arg, "untyped_storage"):
+            return sys.getsizeof(arg.untyped_storage()) < 2**31 - 1
+        return False
+
+    def _add_backend_properties(self, params=None, values=None):
```

Comment:
Nit: move this function as the first so we have the two `@staticmethod` together?